src/package/slate-editor/SlateEditor.js

```diff
@@ -0,0 +1,84 @@
+import { PropTypes } from 'prop-types'
+import React, { Component } from 'react'
+import { Value } from 'slate'
+import classnames from 'classnames'
+
+import { cloneElement } from '../utils/react'
+import { isFunction } from '../utils/type-check'
+import initialEditorState from './initialEditorState'
+
+
+class SlateEditor extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      value: this.migrateStateVersion(props.initialState || initialEditorState),
+      readOnly: false,
+      uid: new Date().getUTCMilliseconds()
+    }
+  }
+
+  //
+  // Migrate Slate's Value object
+  // From v0.25.3
+  // To   v0.31.3
+  //
+  migrateStateVersion (value) {
```

Comment:
If this is going to be common practice for SlateEditor to handle in the future, I think this migration logic should live in its own module and have tests.

Plus, the changes are already not backwards compatible, so why make this part concerned about compatibility?

Reply:
@abobwhite this compatibility function is needed because of the state object generated by previous version of `slate`... We currently save this state object into database and, according to changes on those state objects in this last version of slate, if it was not treated with a compatibility function, like this mentioned on this comment, the state object generated by previous version would break the editor on rendering time...

And, to make a script or a query to normalize all those previous state object on the database, it would require much more effort instead of creating a simple replace function with a regex...

Reply:
Thanks for the explanation. I totally understand the need here. But I'm not sure this belongs within `slate-editor` itself. This seems like something that should be handled by consumers of `slate-editor` for their specific use-cases. And if you insist that it stay in here, could it be broken out into a separate migration utility? That way people could use it by choice vs automatically happening under the hood? And if you still insist that it must remain under the hood, please break it out into a separate file/class and add some tests. This is definitely something that should not go forward without testing around it. It wouldn't be much work to do that.

Reply:
I will decouple this Slate state object compatibility logic from `slate-editor`'s project after this release... I believe it would not impact too much to end users of `slate-editor` because, if the condition to "migrate" the state object does not match, it will do nothing.